libraries/AP_Proximity/AP_Proximity_AirSimSITL.cpp

```diff
@@ -55,29 +57,23 @@ void AP_Proximity_AirSimSITL::update(void)
                 }
             } else {
                 // new sector started, previous one can be updated
-                _distance_valid[_last_sector] = true;
-                _angle[_last_sector] = _angle_deg_last;
-                _distance[_last_sector] = _distance_m_last;
-
+                boundary.mark_distance_valid(true, _last_sector);
+                boundary.set_angle(_angle_deg_last, _last_sector);
+                boundary.set_distance(_distance_m_last, _last_sector);
                 // update boundary
-                update_boundary_for_sector(_last_sector, true);
+                boundary.update_boundary(_last_sector);
+                // update OA database
+                database_push(_angle_deg_last, _distance_m_last);
```

Comment:
at some point we should update this to accept the pitch or the objects won't appear at the correct altitude.

Reply:
Are we talking about the AirSim driver or database_push in general? If its the later, then we already have the rotation matrix taking care of pitch of the vehicle at the backend. Otherwise AirSim, I guess,  is currently only providing horizontal distances ATM.